### ISSUE TITLE
CustomAlias Arg higher $1 not recogniced in TABCOMPLETE in 9.5.0.2

### ISSUE THREAD

**Issue by CloudeLecaw** [Zrips/CMI#7738]
### Description of Issue

Custom alias only show the first argument but no futher arguments anymore
![Snag_801a450](https://user-images.githubusercontent.com/12080017/229326395-859bc318-5ad2-47aa-8cb8-77f1ab45381f.png)

### Version Information

![Snag_75d1ced](https://user-images.githubusercontent.com/12080017/229319260-dd9d88d0-0c6c-482e-9ff0-1ce4d3e9c4c3.png)

### Errors

```shell
none in console
```

###Exeample from Custom Alias
```
  gruppe:
    Cmds:
    - check:$1==null! msg! [playerName] &cDu musst eine Gruppe angeben!\n&b Benutzung&c:&c &c/&cgruppe &c[&cGruppenname&c]&c (Spielername)
    - check:$1!=null! check:$2==null! msg! [playerName] &b&lDu &6warst Rang:&6 &c&l%cmi_user_group% &6und bist jetzt Rang:&6 &2&l$1&6.
    - check:$1!=null! check:$2==null! asPlayer! pex user [playerName] group set $1
    - check:$1!=null! check:$2!=null! msg! [playerName] &b&l$2 &6war Rang:&6 &c&l%parseother_{$2}_{cmi_user_group}% &6und hat jetzt den Rang:&6&2&l$1&6.
    - check:$1!=null! check:$2!=null! msg! $2 &b&l[senderName] &6hat dir den Rang:&6 &2&l$1 &6gegeben.
    - check:$1!=null! check:$2!=null! asPlayer! pex user $2 group set $1
    - delay! 1
    - check:$1!=null! asConsole! cmi tablistupdate
    Perm: true
    Tab: true
    CustomTabs:
    - 'Admin,Builder,CoOwner,Dev,Gast,Gefangener,Guard,Mod,Owner,Senior,Spieler,TestGamer,TestVIP,TuTDone,VIP,VIPV,VIPmini, [playername]'
```

just the first argument is shows. the playername that is set for argument 2 is not shown.
This can be reproduced with a many many custom alias at the moment that have more then 2 arguments.

This was working in 9.5.0.1

Note: Downgrading back to 9.5.0.1 currently solves this issue aggain...

### Agreements

- [X] My server is supported by CMI.
- [X] My version of CMI at the time of this report is up to date.
- [X] I have searched the github and asked around before making this report.

**Comment by mrfloris** [comment]
<img width="966" alt="Screen-Shot 2023-04-02 at 15 05 58" src="https://user-images.githubusercontent.com/28841349/229354680-fddd1dd1-c524-4fa4-9b9a-00dc195ca15c.png">

They seem to all be recognized, when I use check for both null and not null, they all return as expected.

<img width="656" alt="Screen-Shot 2023-04-02 at 15 10 16" src="https://user-images.githubusercontent.com/28841349/229354890-5ed854ce-13f5-4e6c-9763-78a645044453.png">

```yaml
  testingparams:
    Cmds:
    - 'broadcast! Testing params: 0 $0, 1 $1, 2 $2, 3 $3, 4 $4, 5 $5, everything $1-'
    - check:$0==null! broadcast! 0 is null
    - check:$0!=null! broadcast! 0 is not null
    - check:$1==null! broadcast! 1 is null
    - check:$1!=null! broadcast! 1 is not null
    - check:$2==null! broadcast! 2 is null
    - check:$2!=null! broadcast! 2 is not null
    - check:$3==null! broadcast! 3 is null
    - check:$3!=null! broadcast! 3 is not null
    - check:$4==null! broadcast! 4 is null
    - check:$4!=null! broadcast! 4 is not null
    - check:$5==null! broadcast! 5 is null
    - check:$5!=null! broadcast! 5 is not null
    - check:$6==null! broadcast! 6 is null
    - check:$6!=null! broadcast! 6 is not null
    - check:$7==null! broadcast! 7 is null
    - check:$7!=null! broadcast! 7 is not null
```
```
> cmi version
[15:11:03 INFO]: --------------------------------------------------
[15:11:03 INFO]: CMI: 9.5.0.2 SqLite
[15:11:03 INFO]: CMILib: 1.2.5.2
[15:11:03 INFO]: Server: Paper(484) 1.19.4-R0.1-SNAPSHOT
[15:11:03 INFO]: CMI economy: Enabled Vault: 1.7.3-CMI CMI Chat: Enabled
[15:11:03 INFO]: Modules -> 57 enabled 1 disabled: votifier
[15:11:03 INFO]: --------------------------------------------------
> ver
[15:11:05 INFO]: Checking version, please wait...
[15:11:06 INFO]: This server is running Paper version git-Paper-484 (MC: 1.19.4) (Implementing API version 1.19.4-R0.1-SNAPSHOT) (Git: 7a96bf2)
You are running the latest version
Previous version: git-Paper-431 (MC: 1.19.3)
>
```

**Comment by CloudeLecaw** [comment]
well you didnt restated our server -.-
the arg WORKING
just NO tabcomplete

**Comment by CloudeLecaw** [comment]
> well you didnt restated our server -.- you just created a custom alias, tested the args worken after /cmi reload. the args WORKING. we just talk about TABCOMPLET is not suggested
> 
> add tab suggestions to your example. then restart the server so tabcmplete is taken and then try the tab complete if you input the arg all is fine but they are NOT suggested
> 
> ```
>     Tab: true
>     CustomTabs:
>     - '1,2,3,4,5,6, [playername],1,2,3,4,5,6, [playername],1,2,3,4,5,6, [playername],1,2,3,4,5,6, [playername],1,2,3,4,5,6, [playername],1,2,3,4,5,6, [playername],1,2,3,4,5,6'
> ```
> 
> to have 7 suggestions for your commands. (dont use just 1 2 3 4 5 6 etc). to your example and you will see what i ment

well you didnt restated our server -.- you just created a custom alias, tested the args worken after /cmi reload.
the args WORKING. we just talk about TABCOMPLET is not suggested

add tab suggestions to your example. then restart the server so tabcmplete is taken and then try the tab complete
if you input the arg all is fine but they are NOT suggested

    Tab: true
    CustomTabs:
    - '1,2,3,4,5,6, [playername],1,2,3,4,5,6, [playername],1,2,3,4,5,6, [playername],1,2,3,4,5,6, [playername],1,2,3,4,5,6, [playername],1,2,3,4,5,6, [playername],1,2,3,4,5,6'
to have 7 suggestions for your commands. (dont use just 1 2 3 4 5 6 etc). to your example and you will see what i ment

**Comment by CloudeLecaw** [comment]
> <img alt="Screen-Shot 2023-04-02 at 15 05 58" width="966" src="https://user-images.githubusercontent.com/28841349/229354680-fddd1dd1-c524-4fa4-9b9a-00dc195ca15c.png">
> 
> They seem to all be recognized, when I use check for both null and not null, they all return as expected.
> 
> <img alt="Screen-Shot 2023-04-02 at 15 10 16" width="656" src="https://user-images.githubusercontent.com/28841349/229354890-5ed854ce-13f5-4e6c-9763-78a645044453.png">
> 
> ```yaml
>   testingparams:
>     Cmds:
>     - 'broadcast! Testing params: 0 $0, 1 $1, 2 $2, 3 $3, 4 $4, 5 $5, everything $1-'
>     - check:$0==null! broadcast! 0 is null
>     - check:$0!=null! broadcast! 0 is not null
>     - check:$1==null! broadcast! 1 is null
>     - check:$1!=null! broadcast! 1 is not null
>     - check:$2==null! broadcast! 2 is null
>     - check:$2!=null! broadcast! 2 is not null
>     - check:$3==null! broadcast! 3 is null
>     - check:$3!=null! broadcast! 3 is not null
>     - check:$4==null! broadcast! 4 is null
>     - check:$4!=null! broadcast! 4 is not null
>     - check:$5==null! broadcast! 5 is null
>     - check:$5!=null! broadcast! 5 is not null
>     - check:$6==null! broadcast! 6 is null
>     - check:$6!=null! broadcast! 6 is not null
>     - check:$7==null! broadcast! 7 is null
>     - check:$7!=null! broadcast! 7 is not null
> ```
> 
> ```
> > cmi version
> [15:11:03 INFO]: --------------------------------------------------
> [15:11:03 INFO]: CMI: 9.5.0.2 SqLite
> [15:11:03 INFO]: CMILib: 1.2.5.2
> [15:11:03 INFO]: Server: Paper(484) 1.19.4-R0.1-SNAPSHOT
> [15:11:03 INFO]: CMI economy: Enabled Vault: 1.7.3-CMI CMI Chat: Enabled
> [15:11:03 INFO]: Modules -> 57 enabled 1 disabled: votifier
> [15:11:03 INFO]: --------------------------------------------------
> > ver
> [15:11:05 INFO]: Checking version, please wait...
> [15:11:06 INFO]: This server is running Paper version git-Paper-484 (MC: 1.19.4) (Implementing API version 1.19.4-R0.1-SNAPSHOT) (Git: 7a96bf2)
> You are running the latest version
> Previous version: git-Paper-431 (MC: 1.19.3)
> >
> ```

well you didnt restated our server -.- and you added no tabcomplete.  you just created a custom alias, tested the args worken after /cmi reload.
the args WORKING. we just talk about TABCOMPLET is not suggested

add tab suggestions to your example. then restart the server so tabcmplete is taken and then try the tab complete
if you input the arg all is fine but they are NOT suggested

```
    Tab: true
    CustomTabs:
    - '1,2,3,4,5,6, [playername],1,2,3,4,5,6, [playername],1,2,3,4,5,6, [playername],1,2,3,4,5,6, [playername],1,2,3,4,5,6, [playername],1,2,3,4,5,6, [playername],1,2,3,4,5,6'
```

to have 7 suggestions for your commands. (dont use just 1 2 3 4 5 6 etc). to your example and you will see what i ment

**Comment by mrfloris** [comment]
I do not know what you mean with "restated our server" if that is supposed to say perhaps "restarted your server", yes, I did /stop, and start, I don't use /reload or anything. The first screenshot shows me creating the alias, the second one is after the restart. 

The title is "CustomAlias Arg higher $1 not recogniced in 9.5.0.2", it mentioned nothing about tab complete. Reading your post it was honestly unclear to me that with argument not recognized, you meant 'custom tab completes'. That explains the first screenshot you posted a bit better. 

Okay, since I understand you better now, I can continue testing ~ I remember posting to zrips in a DM that this wasn't working, so I suspect this is indeed a bug. 

(tnx for changing the ticket title, it makes more sense now).

Thank you for the additional information.

**Comment by mrfloris** [comment]
<img width="508" alt="Screen-Shot 2023-04-02 at 16 19 50" src="https://user-images.githubusercontent.com/28841349/229358708-eb31345f-3c12-4e97-b016-db3871d89e97.png">
This is how it shows to me, some seem to work, when a comma is involved, they do not. (seemingly)

**Comment by Truongvv10** [comment]
Can confirm, I'm having this issue too.

**Comment by CloudeLecaw** [comment]
> Can confirm, I'm having this issue too.

at the moment only a downgrade to 9.5.0.1 solved this

So lets hope for a fix may with 9.5.0.3 :3

**Comment by CloudeLecaw** [comment]
Issue still exist in 9.5.0.3

So lets hope for a fix may with 9.5.0.4 :3

**Comment by Zrips** [comment]
Will be fixed with next version
